### DISCUSSION TITLE
Best way to temporarily connect two objects in a rigid way (PyBullet)?

### DISCUSSION THREAD

**Discussion by narittt** [bulletphysics/bullet3#4483] [Q&A] [answered]
Hi all,

I am trying to implement a simulation where the user can press a button to "pick up" an object, which I want to just essentially be attaching one object to another. I also want them to be able to place an object, or detach the two. I have one cylinder object that is a multi body and one object loaded in from a URDF file.

I saw the joinURDF.py example in pybullet utils and have also tried with constraints. However, there is a delay with the constraint as the attachment does not seem to be rigid. I was wondering what would be the simplest way to implement joinURDF as well as the opposite (detach the two)?

Thank you

**Comment by Avithohol** [comment, marked as answer]
Try to figure out the point in 3d space where the body gets picked by, then create a simple point to point constraint and keep it while the body is "picked up":

				btVector3 localPivot = body->getCenterOfMassTransform().inverse() * pickPos;
				mPickConstraint = std::make_unique<btPoint2PointConstraint>(*body, localPivot);
				mDynamicsWorld->addConstraint(mPickConstraint.get(), true);
				mPickConstraint->m_setting.m_impulseClamp = 3.0f;
				mPickConstraint->m_setting.m_tau = 0.01f; //weak constarint

if you wanna grab the body around by the picking point, then you need to change the frame:
```
glm::vec3 pivotPoint = oldPos + newPos;
btVector3 newPivotB(pivotPoint.x, pivotPoint.y, pivotPoint.z);
mPickConstraint->setPivotB(newPivotB);

```
hope that helps!

**Reply by narittt** [reply]
Thanks! I am using PyBullet so I used a similar implementation with createConstraint(). Had the object picking up be the parent frame and the other one be the child, with removeConstraint() to  drop the object.

**Reply by Ghassan-Nafash** [reply]
Hallo @narittt , i have found this blog because i want actually to do the same approach, so that the robot EE should pick an Object, i have tried the function creatConstraint(), but wrong motion happens. Can you explain me more or if you could share the code? thanks

**Reply by Ghassan-Nafash** [reply]
by the way i am implementing it in python :)
